cmd/icinga-kubernetes/main.go

```diff
@@ -51,9 +46,37 @@ func main() {
 
 		return nil
 	})
+
+	flag.StringVar(&contextName, "context", "", "name of the kubeconfig context to use")
```

Comment:
Is `context` and `cluster` really necessary?

I have something like this in my stash:

```
	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
	overrides := clientcmd.ConfigOverrides{}
	kflags := clientcmd.RecommendedConfigOverrideFlags("")
	flag.StringVar(&loadingRules.ExplicitPath, "kubeconfig", "", "Path to a kube config. Only required if out-of-cluster")
	clientcmd.BindOverrideFlags(&overrides, flag.CommandLine, kflags)
	flag.Parse()
	clientconfig := clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, &overrides, os.Stdin)
```

Reply:
You are absolutely right! I have now also completely changed to ‘pflag’ (cause BindOverrideFlags needs a *pflag.FlagSet) and adapted the version-flag so that it also works with it